### PR TITLE
Expose `TweaksConfig.forceReuseVideoCodecReasons` from Android

### PR DESCRIPTION
**Pull request by saravanans-github**
## Description
Expose Android SDKs TweaksConfig.forceReuseVideoCodecReasons
## Checklist
- [x] 🗒 `CHANGELOG` entry

### CODE REVIEW

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt [hunk 1]
```diff
@@ -189,6 +200,16 @@ fun ReadableMap.toTweaksConfig(): TweaksConfig = TweaksConfig().apply {
...
+    withStringArray("forceReuseVideoCodecReasons") {
+        val mutableSet = mutableSetOf<ForceReuseVideoCodecReason>()
+        for (item in it) {
+            val reason = item?.forceReuseVideoCodecReason()
+            if (reason != null) {
+                mutableSet.add(reason)
+            }
+        }
+        forceReuseVideoCodecReasons = mutableSet
+    }
```
Comment (krocard) [inline android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt:212]:
```suggestion
        forceReuseVideoCodecReasons = it.mapNotNull(String::forceReuseVideoCodecReason)
```

File: src/tweaksConfig.ts [hunk 2]
```diff
@@ -170,4 +188,14 @@ export interface TweaksConfig {
    * @platform iOS
    */
   updatesNowPlayingInfoCenter?: boolean;
+
+  /**
+   * When switching between video formats (eg: adapting between video qualities) the codec might be recreated due to several reasons.
+   * This behaviour can cause brief black screens when switching between video qualities as codec recreation can be slow.
+   *
```
Comment (krocard) [inline src/tweaksConfig.ts:195]:
```suggestion
   *
   * If a device is know to support video format changes and keep the current decoder without issues,
   * this set can be filled with multiple `ForceReuseVideoCodecReason` and avoid the black screen.
   *
```

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt [hunk 3]
```diff
@@ -167,6 +168,16 @@ fun ReadableMap.toStyleConfig(): StyleConfig = StyleConfig().apply {
     withString("scalingMode") { scalingMode = ScalingMode.valueOf(it) }
 }
 
+/**
+ * Converts any JS string into an `AdSourceType` enum value.
+ */
+private fun String.forceReuseVideoCodecReason(): ForceReuseVideoCodecReason? = when (this) {
```
Comment (krocard) [inline android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt:174]:
```suggestion
private fun String.toForceReuseVideoCodecReason(): ForceReuseVideoCodecReason? = when (this) {
```

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt [hunk 4]
```diff
@@ -167,6 +168,16 @@ fun ReadableMap.toStyleConfig(): StyleConfig = StyleConfig().apply {
     withString("scalingMode") { scalingMode = ScalingMode.valueOf(it) }
 }
 
+/**
+ * Converts any JS string into an `AdSourceType` enum value.
```
Comment (krocard) [inline android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt:172]:
```suggestion
 * Converts any JS string into an `ForceReuseVideoCodecReason` enum value.
```

**Review by krocard — CHANGES_REQUESTED** [review]
I didn't test the PR, just reviewed the code change.

**Review by matamegger — CHANGES_REQUESTED** [review]

**Comment by strangesource** [comment]
Thanks for the reviews, I'll pick this up tomorrow.

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt (hunk 1 above)
Comment (strangesource) [reply android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt]: [6c0fe0b](https://github.com/bitmovin/bitmovin-player-react-native/pull/561/commits/6c0fe0be72537d83455ef8ef5378a17d315bcd97)

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt (hunk 3 above)
Comment (strangesource) [reply android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt]: [6c0fe0b](https://github.com/bitmovin/bitmovin-player-react-native/pull/561/commits/6c0fe0be72537d83455ef8ef5378a17d315bcd97)

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt (hunk 4 above)
Comment (strangesource) [reply android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt]: [6c0fe0b](https://github.com/bitmovin/bitmovin-player-react-native/pull/561/commits/6c0fe0be72537d83455ef8ef5378a17d315bcd97)

File: android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (strangesource) [reply android/src/main/java/com/bitmovin/player/reactnative/converter/JsonConverter.kt]: [6c0fe0b](https://github.com/bitmovin/bitmovin-player-react-native/pull/561/commits/6c0fe0be72537d83455ef8ef5378a17d315bcd97)

File: src/tweaksConfig.ts [hunk 5]
```diff
@@ -170,4 +188,14 @@ export interface TweaksConfig {
...
+   * Default is `null` i.e not set
+   *
+   * @platform Android
+   */
```
[the remark this answers is not in the document]
Comment (strangesource) [reply src/tweaksConfig.ts]: [0ec9d98](https://github.com/bitmovin/bitmovin-player-react-native/pull/561/commits/0ec9d9803d26f4617e9aa96bcc608edf4864c50c)

File: src/tweaksConfig.ts [hunk 6]
```diff
@@ -1,3 +1,21 @@
...
+  /**
+   * The new video quality exceed the decoder's configured maximum sample size.
+   */
+  MaxInputSizeExceeded = 'MaxInputSizeExceeded',
+  /**
+   * The new video quality exceed the decoder's configured maximum resolution.
+   */
+  MaxResolutionExceeded = 'MaxResolutionExceeded',
+}
+
```
[the remark this answers is not in the document]
Comment (strangesource) [reply src/tweaksConfig.ts]: [0ec9d98](https://github.com/bitmovin/bitmovin-player-react-native/pull/561/commits/0ec9d9803d26f4617e9aa96bcc608edf4864c50c)

File: src/tweaksConfig.ts (hunk 2 above)
Comment (strangesource) [reply src/tweaksConfig.ts:206]: [0ec9d98](https://github.com/bitmovin/bitmovin-player-react-native/pull/561/commits/0ec9d9803d26f4617e9aa96bcc608edf4864c50c)

**Review by matamegger — APPROVED** [review]
@strangesource Please revisit the Changelog entry

**Review by krocard — APPROVED** [review]

**Pull request merged**